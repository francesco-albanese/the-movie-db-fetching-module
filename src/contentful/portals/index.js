import { 
  contentTypes,
  contentfulClient, 
  urls 
} from '#contentful/contentful.const'
import { transformPortalResponse } from '#transformation'
import { parseStringifiedData } from '#utils'

export const getPortals = (locale = '*') => {
  return contentfulClient.get(urls.entries, {
    params: {
      content_type: contentTypes.portal,
      include: 3,
      locale,
      select: 'sys.id,fields.name,fields.sections'
    },
    transformResponse: [
      data => parseStringifiedData(data),
      data => transformPortalResponse(data)
    ]
  })
    .then(data => data)
}