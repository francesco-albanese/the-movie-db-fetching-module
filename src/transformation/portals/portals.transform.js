import { isEmpty } from 'lodash-es'
import { getFirstObjectKey, getSections } from '#utils'

export const transformPortalResponse = data => {
  if (isEmpty(data)) {
    return []
  }

  const { items, includes } = data

  return items.map(portal => {

    const { 
      name,
      sections 
    } = portal.fields
    
    const parsedSections = !isEmpty(sections) 
      ? getSections(portal, includes.Entry)
      : null

    const transformedPortals = {
      name: name[ getFirstObjectKey(name) ],
      sections: parsedSections
    }
    
    return transformedPortals
  })

}