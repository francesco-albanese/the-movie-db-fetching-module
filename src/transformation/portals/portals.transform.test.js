import { transformPortalResponse } from './portals.transform'
import { data } from './__fixtures__/portals.transform.fixtures'

describe('transformPortalResponse', () => {
  it('returns an empty array if data is missing', () => {
    expect(transformPortalResponse()).toEqual([])
  })

  it('returns transformed portals', () => {
    const result = transformPortalResponse(data)
    expect(result).toMatchSnapshot()
  })
})