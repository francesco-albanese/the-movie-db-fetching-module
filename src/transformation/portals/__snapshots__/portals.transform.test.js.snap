// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformPortalResponse returns transformed portals 1`] = `
Array [
  Object {
    "name": "MainMenu",
    "sections": Object {
      "FavouritesLink": Object {
        "Favourites": Object {
          "icon": Object {
            "en": "Favorite",
          },
          "label": Object {
            "en": "My Favourites",
            "it": "Miei Preferiti",
          },
          "name": Object {
            "en": "Favourites",
          },
          "path": Object {
            "en": "/en/favourites",
            "it": "/it/favoriti",
          },
          "reference": Object {
            "en": "favourites",
          },
        },
      },
      "HomeLink": Object {
        "Home": Object {
          "icon": Object {
            "en": "Home",
          },
          "label": Object {
            "en": "Home",
            "it": "Home",
          },
          "name": Object {
            "en": "Home",
          },
          "path": Object {
            "en": "/en",
            "it": "/it",
          },
          "reference": Object {
            "en": "home",
          },
        },
      },
      "LanguageSelector": Object {
        "LanguageSelectorText": Object {
          "lineOne": Object {
            "en": "Select Language",
            "it": "Seleziona Lingua",
          },
          "lineTwo": Object {
            "en": "Selected",
            "it": "Selezionata",
          },
          "name": Object {
            "en": "LanguageSelectorText",
          },
        },
      },
      "MainMenu": Object {
        "NowPlaying": Object {
          "icon": Object {
            "en": "PlayArrow",
          },
          "label": Object {
            "en": "Now Playing",
            "it": "Adesso In Onda",
          },
          "name": Object {
            "en": "NowPlaying",
          },
          "reference": Object {
            "en": "now-playing",
          },
        },
        "Popular": Object {
          "icon": Object {
            "en": "LocalMovies",
          },
          "label": Object {
            "en": "Popular Movies",
            "it": "Film Popolari",
          },
          "name": Object {
            "en": "Popular",
          },
          "reference": Object {
            "en": "popular",
          },
        },
        "TopRated": Object {
          "icon": Object {
            "en": "StarRate",
          },
          "label": Object {
            "en": "Top Rated",
            "it": "Premiatissimi",
          },
          "name": Object {
            "en": "TopRated",
          },
          "reference": Object {
            "en": "top-rated",
          },
        },
        "Upcoming": Object {
          "icon": Object {
            "en": "RecentActors",
          },
          "label": Object {
            "en": "Upcoming",
            "it": "In Arrivo",
          },
          "name": Object {
            "en": "Upcoming",
          },
          "reference": Object {
            "en": "upcoming",
          },
        },
      },
    },
  },
]
`;
